File: 2892-check-if-array-is-good/check-if-array-is-good.py
class Solution:
    def isGood(self, nums: List[int]) -> bool:
        max_value = max(nums)
        n = len(nums)

        if (max_value+1 != n):
            return False
        nums.sort()
        for i in range(n-1):
            if (i+1!=nums[i]):
                return False
        return nums[n-2]==nums[n-1]
        

File: 2892-check-if-array-is-good/check-if-array-is-good.cpp
class Solution {
public:
    bool isGood(vector<int>& nums) {
        int n = nums.size();
        vector<int> cnt(210,0);

        for(int&a: nums){
            cnt[a]++;
        }

        for(int i=1; i<=n-2;++i){
            if (cnt[i]!=1)
                return false;
        }

        return cnt[n-1]==2 ? true:false;
        
    }
};